pkg/provisioner/ironic/ironic.go

```diff
@@ -625,102 +641,51 @@ func (p *ironicProvisioner) getUpdateOptsForNode(ironicNode *nodes.Node) (update
 				p.host.HardwareProfile()))
 	}
 
-	// image_source
-	var op nodes.UpdateOp
-	if _, ok := ironicNode.InstanceInfo["image_source"]; !ok {
-		op = nodes.AddOp
-		p.log.Info("adding image_source")
-	} else {
-		op = nodes.ReplaceOp
-		p.log.Info("updating image_source")
-	}
+	// instance_uuid
+	p.log.Info("setting instance_uuid")
 	updates = append(
 		updates,
 		nodes.UpdateOperation{
-			Op:    op,
-			Path:  "/instance_info/image_source",
-			Value: p.host.Spec.Image.URL,
+			Op:    nodes.ReplaceOp,
+			Path:  "/instance_uuid",
+			Value: string(p.host.ObjectMeta.UID),
 		},
 	)
 
-	checksum, checksumType, _ := p.host.GetImageChecksum()
-
-	// image_os_hash_algo
-	if _, ok := ironicNode.InstanceInfo["image_os_hash_algo"]; !ok {
+	// cpu_arch
+	//
+	// FIXME(dhellmann): This should come from inspecting the
+	// host.
```

Comment:
We're now calling this func before inspection, which will make it hard to change this.

Reply:
It's not clear to me why we need to set this at all if it can come from inspection, but when we do drop the profile we would necessarily not set it here unless we did have inspection data.